src/oscar/apps/catalogue/abstract_models.py

```diff
@@ -495,6 +495,16 @@ def options(self):
         pclass_options = self.get_product_class().options.all()
         return set(pclass_options) or set(self.product_options.all())
 
+    @property
+    def has_options(self):
```

Comment:
Please add unit tests for this logic, and also to verify that the queryset gets correctly annotated.

Reply:
Done.